src/main/java/org/commcare/formplayer/util/FormplayerSentryUserProvider.java

```diff
@@ -0,0 +1,24 @@
+package org.commcare.formplayer.util;
+
+import io.sentry.protocol.User;
+import io.sentry.spring.SentryUserProvider;
+import org.commcare.formplayer.beans.auth.HqUserDetailsBean;
+import org.springframework.stereotype.Component;
+
+@Component
+class FormplayerSentryUserProvider implements SentryUserProvider {
+    public User provideUser() {
+        User user = new User();
+        FormplayerHttpRequest request = RequestUtils.getCurrentRequest();
```

Comment:
This is a really helpful pattern that I don't think I've seen, is this why the component itself isn't Request scoped? 

I don't understand the Spring component selection process super well, but is there an alternative pattern where a request scoped UserProvider takes precedence over a singleton provider?

Reply:
I don't think Sentry would support more than one user provider. This isn't request scoped since you can report errors outside of the request scope.

Reply:
Ah, sorry, I think I framed that poorly, I was assuming that the second provider would be the one that was used outside of the request scope.

IE: You have an `@Bean` autowired: If you are in a request context, the component that gets injected is one which presumes there's a request and grabs the user, while if you are out of a request context it provides a bean that doesn't have that info. Basically using the Dependency Injection scheme as a kind of Polymorphism. 

As far as I've read, the expectation is pretty much just that different beans are really only provided in different runtime contexts (IE: "Are you in a test environment or a prod environment"), but I wasn't sure if it was also used for different lifecycle contexts.

Reply:
I'm not sure - I haven't dug into that. I think the one disadvantage of doing that is it would create 2 classes instead of 1 for basically doing the same thing. I guess we could put them both in a Kotlin file in which case they source would be together. 

I'll make a note to look into that